src/main/java/seedu/address/model/person/Priority.java

```diff
@@ -0,0 +1,70 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Food item priority level in the food inventory system.
+ * Guarantees: immutable; is valid as declared in {@link #isValidPriority(String)}
+ */
+public class Priority {
+    public enum Level {
+        LOW,
+        MEDIUM,
+        HIGH
+    }
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Priority should be either high, medium or low.";
+    public final Level value;
+
+    /**
+     * Constructs a {@code Priority}.
+     *
+     * @param priority A valid priority level.
+     */
+    public Priority(String priority) {
+        priority = priority.toLowerCase();
+        requireNonNull(priority);
+        checkArgument(isValidPriority(priority), MESSAGE_CONSTRAINTS);
+        switch(priority) {
+        case "low":
+            value = Level.LOW;
+            break;
+        case "medium":
+            value = Level.MEDIUM;
+            break;
+        case "high":
+            value = Level.HIGH;
+            break;
+        default:
+            throw new IllegalArgumentException(MESSAGE_CONSTRAINTS);
+        }
+    }
+
+    /**
+     * Returns true if a given string is a valid priority number.
```

Comment:
This should be "Returns true if a given string is a valid priority level"

Reply:
Ok noted. Will change.